server/index.js

```diff
@@ -0,0 +1,38 @@
+var express = require('express');
+var path = require('path');
+
+// require routes
+var events = require('./routes/eventRoutes');
+var users = require('./routes/userRoutes');
+var groups = require('./routes/groupRoutes');
+
+// require middleware
+var bodyParser = require('body-parser');
+var morgan = require('morgan');
+var expressSession = require('express-session');
+var passport = require('passport');
+
+const port = process.env.PORT || 8000;
+
+var app = express();
+app.use(passport.initialize());
+app.use(passport.session());
+
+// setup extra middleware
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
+
+app.use('/api/users', users);
+app.use('/api/events', events);
+app.use('/api/groups', groups);
+
+
+app.use((req, res, next)=> {
+  var err = new Error('ERROR 404 Sorry can\'t find what you\'re looking for!');
+  err.status = 404;
+  next(err);
+});
+
```

Comment:
you have a thing to catch 404s, but it uses next to pass the error to an error handler when currently there is none. Add another route after the 404 route that sends the client the error whenever `next(err)` is called.

``` javascript
app.use((err, req, res, next) => {
  var status = err.status || 500;
  res.status(status).send(err.message);
};
```